docs/spec_draft.md

```diff
@@ -609,6 +610,44 @@ matrix, then the behavior is undefined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.complex
+
+### Semantics
+
+Performs element-wise conversion to a complex value from a pair of real and
+imaginary values, `lhs` and `rhs`, and produces a `result` tensor.
+
+### Inputs
+
+| Name  | Type                          |
+|-------|-------------------------------|
+| `lhs` | tensor of type `f32` or `f64` |
+| `rhs` | tensor of type `f32` or `f64` |
+
+### Outputs
+
+| Name     | Type                   |
+|----------|------------------------|
+| `result` | tensor of complex type |
+
+### Constraints
+
+  * (C1) `lhs` and `rhs` have the same floating-point element type.
+  * (C2) shape(`lhs`) $=$ shape(`rhs`) $=$ shape(`result`).
+  * (C3) element_type(`lhs`) $=$ underlying floating-point element type in
+    element_type(`result`).
```

Comment:
Also, not sure if it's just me, but it feels more natural to express C2 and C3 `with element_type(result)` on the left-hand side of the equation, i.e.:

```
  * (C2) `shape(result) = shape(lhs)`.
  * (C3) `element_type(result) = complex_type(element_type(lhs))`.
```

This way, it's also easier to implement type inference for this op based on the Constraints section. It's not hard to invert `element_type(lhs) = real_type(element_type(result))` - but it feels awkward. Given how many times I wrote "feels", this is not a blocking comment, and we can leave figuring this out to #289.